### PR TITLE
Rename secret across application - Closes #1994

### LINKED ISSUE #1994: Rename secret across application
Follow up to #1965 

### Expected behavior

We should use `passphrase`, `password`, `key` etc consistently.

### Actual behavior

We use `secret` in some places to denote different things. E.g. 
- `config.json` (`forging.secret` should be `forging.delegates`?)
- `test/data/config.json`
- `test/data/genesis_delegates.json`
- `test/functional/http/get/delegates.js`
- `test/functional/modules/blocks/verify.js`
- `test/functional/system/blocks/process/on_receive_block.js`
- `test/unit/logic/transaction.js`

### Which version(s) does this affect? (Environment, OS, etc...)

1.0.0

### PR DESCRIPTION
**Pull request by 4miners**
### What was the problem?
We use secret in some places to denote different things. E.g.
### How did I fix it?
Rename:
- `secret` to `delegates` in `config.json`
- `secret` to `passphrase`
- `secondSecret` to `secondPassohrase`
- `password` to `passphrase` when used in `passphrase` context
- other things
### How to test it?
Run test suite.
### Review checklist

* The PR solves #1994
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: logger.js [hunk 1]
```diff
@@ -58,9 +58,9 @@ module.exports = function(config) {
 		config.errorLevel = errorLevel;
 	};
 
-	function snipsecret(data) {
+	function snipFragileData(data) {
 		for (var key in data) {
-			if (key.search(/secret/i) > -1) {
+			if (key.search(/passphrase|password/i) > -1) {
```
Comment (MaciejBaj) [inline logger.js]: Why password here?

Comment (4miners) [reply logger.js]: Because `password` is also sensitive data, same as `passphrase`.

Comment (MaciejBaj) [reply logger.js]: Same as secret then.

Comment (MaciejBaj) [reply logger.js]: But unused.

**Pull request merged**